src/System.IO.FileSystem/tests/DirectoryInfo/ToString.cs

```diff
@@ -1,141 +1,42 @@
-                iCountErrors++;
-                printerr("Error_209xu! Incorrect Directory returned , dir==" + dir.ToString());
-                Console.WriteLine(dirName);
-                Console.WriteLine(dir == null);
-                Console.WriteLine(dir);
-            }
-            // [] Root drive
-
-            strLoc = "Loc_20u9x";
-
-            dir = new DirectoryInfo(Path.GetPathRoot(Directory.GetCurrentDirectory()));
-            iCountTestcases++;
-            if (!dir.ToString().Equals(Path.GetPathRoot(Directory.GetCurrentDirectory())))
-            {
-                iCountErrors++;
-                printerr("Error_2098x! Incorrect dir returned==" + dir.ToString());
-            }
-
-            string testDir = Path.Combine(dirName, "TestDir");
-            Directory.CreateDirectory(testDir);
-            dir = new DirectoryInfo(testDir);
-            iCountTestcases++;
-
-            if (!dir.FullName.Equals(testDir))
-            {
-                iCountErrors++;
-                printerr("Error_298yx! Incorrect dir constructed, dir==" + dir.ToString());
-            }
-            dir.Delete();
-
-            // [] Whitespace string
-
-            strLoc = "Loc_298yb";
-
-            if (Interop.IsWindows) // whitespace-only names are valid on Unix
-            {
-                iCountTestcases++;
-                try
-                {
-                    dir = new DirectoryInfo("      ");
-                    dir.Create();
-                    iCountErrors++;
-                    printerr("Error_09rux! Expected exception not thrown, dir==" + dir.ToString());
-                }
-                catch (ArgumentException)
-                {
-                }
-                catch (Exception exc)
-                {
-                    iCountErrors++;
-                    printerr("Bug 14866 Error_4577c! Incorrect exception thrown, exc==" + exc.ToString());
-                }
-            }
-
-            // [] Go for same dir
-
-            strLoc = "Loc_949gg";
-
-            dir = new DirectoryInfo(".");
-            iCountTestcases++;
-            if (!dir.FullName.Equals(Directory.GetCurrentDirectory()))
-            {
-                iCountErrors++;
-                printerr("Error_299xu! Incorrect dir constructed, dir==" + dir.ToString());
-            }
-
-            // [] go one dir up
-
-            strLoc = "Loc_9937a";
-
-            dir = new DirectoryInfo("..");
-            iCountTestcases++;
-            if (!dir.FullName.Equals(Path.GetDirectoryName(Directory.GetCurrentDirectory())))
-            {
-                iCountErrors++;
-                printerr("Error_298xy! Incorrect dir returned, dir==" + dir.ToString());
-            }
+            string path = GetTestFilePath();
+            var info = new DirectoryInfo(path);
+            Assert.Equal(Path.GetFileName(path), info.ToString());
         }
-        catch (Exception exc_general)
+
+        [Fact]
+        public void RootDrive()
```

Comment:
Print the root of the current drive. Perhaps "RootOfCurrentDrive" or "Root" would be better.